src/pipelines/epidemiology/co_authority.py

```diff
@@ -60,18 +61,19 @@ def parse_dataframes(
         value_columns = ["new_confirmed", "new_deceased", "new_recovered"]
         for value_column in value_columns:
             subset = data.rename(columns={"date_{}".format(value_column): "date"})[index_columns]
-            subset = subset[~subset.date.isna() & (subset.date != "-   -")].dropna()
+            # Parse dates to ISO format.
+            subset.date = subset.date.apply(safe_datetime_parse)
+            # Some dates are badly formatted as 31/12/1899 in the raw data
+            # we can drop these.
+            subset = subset[(subset.date != datetime(1899, 12, 31))].dropna()
```

Comment:
Is there a reason why you are converting back to datetime objects to make this comparison? I wonder if we can filter out the bad dates in the `merged` dataframe so we can perform the operation just once and without converting back to datetime.

Reply:
Given the number of bad dates I thought doing it in the subset/merge operation would be overall more efficient since it reduces the number of rows in outer join. The conversion to datetime was already happening later in merged, I just moved it earlier. My general heuristic with dates is always parse then compare, otherwise badly formatted date input slips past.

Reply:
Fair enough, LGTM then.